crates/turborepo-ui/src/prefixed.rs

```diff
@@ -14,58 +14,84 @@ use crate::UI;
 /// want for replaying logs.
 pub struct PrefixedUI<W> {
     ui: UI,
-    output_prefix: StyledObject<String>,
-    warn_prefix: StyledObject<String>,
+    output_prefix: Option<StyledObject<String>>,
+    warn_prefix: Option<StyledObject<String>>,
+    error_prefix: Option<StyledObject<String>>,
     out: W,
     err: W,
+    default_prefix: StyledObject<String>,
 }
 
 impl<W: Write> PrefixedUI<W> {
-    pub fn new(
-        ui: UI,
-        output_prefix: StyledObject<String>,
-        warn_prefix: StyledObject<String>,
-        out: W,
-        err: W,
-    ) -> Self {
+    pub fn new(ui: UI, out: W, err: W) -> Self {
         Self {
             ui,
-            output_prefix,
-            warn_prefix,
             out,
             err,
+            output_prefix: None,
+            warn_prefix: None,
+            error_prefix: None,
+            default_prefix: Style::new().apply_to(String::new()),
         }
     }
 
+    pub fn with_output_prefix(mut self, output_prefix: StyledObject<String>) -> Self {
+        self.output_prefix = Some(self.ui.apply(output_prefix));
```

Comment:
Note: we now eagerly apply the `ui` to the prefixes when they get passed in. Since we don't offer a way to change `ui` after construction we don't need to wait until the actual line writing to perform this operation.